app/services/setting_registry.rb

```diff
@@ -0,0 +1,100 @@
+class SettingRegistry
+  include Singleton
+
+  def initialize
+    @settings = {}
+    load
+  end
+
+  def logger
+    Foreman::Logging.logger('app')
+  end
+
+  def find(name)
+    @settings[name.to_s]
+  end
+
+  # Returns a setting effective value
+  # Call as `Foreman.settings[<name>]`
+  #
+  #   Foreman.settings['default_locale'] => nil.
+  #   Foreman.settings[:default_locale] => nil.
+  #
+  def [](name)
+    definition = find(name)
+    logger.warn("Setting #{name} has no definition, please define it before using") and return unless definition
+    definition.value
+  end
+
+  def []=(name, value)
+    definition = find(name)
+    raise ::Foreman::Exception.new(N_("Setting definition for '%s' not found, can not set"), name) unless definition
+    db_record = _find_or_new_db_record(name)
+    db_record.value = value
+    db_record.save!
+    definition.value = db_record.value
+  end
+
+  def load
+    # add() all setting definitions
+    Setting.descendants.each do |cat_cls|
+      if cat_cls.default_settings.empty?
+        # Setting category uses really old way of doin things
+        _load_category_from_db(cat_cls)
+      else
+        cat_cls.default_settings.each { |s| _add(s[:name], s.except(:name).merge(category: cat_cls.name)) }
+      end
+    end
+
+    # load all db values
+    load_values
+
+    # if create_missing create missing values in DB
+    # not needed now as old load mechanism takes care of that
+  end
+
+  def load_values
+    loaded_names = []
+    Setting.unscoped.all.each do |set|
+      set_definition = find(set.name)
+      logger.warn("Setting #{set.name} has no definition, clean up your database") and next unless set_definition
+      loaded_names << set.name
+      set_definition.updated_at = set.updated_at
+      set_definition.value = set.value
+    end
+    # load nil to set value to default
+    @settings.except(*loaded_names).each do |name, set_definition|
```

Comment:
freaking asterisk

Reply:
sorry, my bad, i suggested this change :)

Reply:
And I've blindly applied it without thinking, not really your bad :)
I remember struggling with exactly this few times already, I'm even thinking if the method should warn about that xD